File: examples/java/StringUtils.java
package examples.java;

import com.google.common.base.Joiner;
import com.google.common.base.Splitter;
import com.google.common.collect.ImmutableList;
import java.util.List;

/**
 * Utility library using Guava
 */
public class StringUtils {

    public static List<String> splitWords(String text) {
        return ImmutableList.copyOf(
            Splitter.on(' ')
                .trimResults()
                .omitEmptyStrings()
                .split(text)
        );
    }

    public static String joinWithCommas(List<String> items) {
        return Joiner.on(", ").join(items);
    }

    public static String reverseWords(String text) {
        List<String> words = splitWords(text);
        return joinWithCommas(ImmutableList.copyOf(words).reverse());
    }
}

File: examples/python/string_utils.py
"""String utility functions demonstrating library code."""

from typing import List
import re


class StringUtils:
    """Utility class for string operations."""

    @staticmethod
    def split_words(text: str) -> List[str]:
        """Split text into words."""
        # Use regex to split on whitespace and punctuation
        words = re.findall(r'\b\w+\b', text)
        return words

    @staticmethod
    def reverse_words(text: str) -> str:
        """Reverse the order of words in text."""
        words = text.split()
        return ' '.join(reversed(words))

    @staticmethod
    def join_with_commas(words: List[str]) -> str:
        """Join words with commas."""
        return ', '.join(words)

    @staticmethod
    def capitalize_words(text: str) -> str:
        """Capitalize the first letter of each word."""
        return ' '.join(word.capitalize() for word in text.split())

    @staticmethod
    def count_vowels(text: str) -> int:
        """Count the number of vowels in text."""
        vowels = 'aeiouAEIOU'
        return sum(1 for char in text if char in vowels)

    @staticmethod
    def is_palindrome(text: str) -> bool:
        """Check if text is a palindrome (ignoring case and spaces)."""
        cleaned = ''.join(text.lower().split())
        return cleaned == cleaned[::-1]